### PR TITLE
[tidy] Misc fixes to User and Match services

### PR DESCRIPTION
**Pull request by willjvsmith**
* Assigns a fixed version to Kong docker image
* Reorders functions/structs to follow CRUD
* Adds an ignore flag, `valid:"-"`, to Match structs
  * This means govalidator's `validateStruct` can be called on the struct without causing an error
  * Though we don't need to call `validateStruct` on them anyway, so could take out (but would mean more effort code generating later"
* Forces calls to CREATE and UPDATE to pass all fields
  * Makes CREATE and UPDATE request struct fields pointers, so it can be discerned if they were omitted (otherwise they take zero-value)
* Renames `Users` table to `User_Temple`
  * Avoids pluralising problem, e.g. `User` -> `Users`, `Match` -> `Matches`
  * Unfortunately `User` is reserved in Postgres, `Temple` suffix handles this
    * Decision made to maintain a map of reserved names when generating
* Removes User and Match CREATE endpoints returning the row, returns `{}` to match UPDATE
* Change Match UPDATE to be a HTTP PUT rather than PATCH
* Rename `ExtractUserIDFromRequest` in User service to `ExtractIDFromRequest`, to match the same function in Match service (less code to generate)

The diff here is not the best because of the reordering of functions/structs to follow CRUD. If too hard to follow LMK and I'll put the reordering in a separate PR. I've tried to detail everything above to mitigate.

#### Test plan

```
> ./build.sh
> sh kong/configure-kong.sh
```

User service:
```
> curl -X GET localhost:8000/api/user/1
{"error":"user not found with ID 1"}
> curl -X POST localhost:8000/api/user -d '{"name":"Will"}'
{}
> curl -X GET localhost:8000/api/user/1
{"ID":1,"Name":"Will"}
> curl -X PUT localhost:8000/api/user/1 -d '{"name":"Charlie"}'
{}
> curl -X GET localhost:8000/api/user/1
{"ID":1,"Name":"Charlie"}
> curl -X DELETE localhost:8000/api/user/1
{}
> curl -X GET localhost:8000/api/user/1
{"error":"user not found with ID 1"}
```

Match service:
```
> curl -X GET localhost:8000/api/match/1
{"error":"match not found with ID 1"}
> curl -X POST localhost:8000/api/match -d '{"UserOne":1, "UserTwo":2}'
{}
> curl -X GET localhost:8000/api/match/1
{"ID":1,"UserOne":1,"UserTwo":2,"MatchedOn":<TIME>}
> curl -X PUT localhost:8000/api/match/1 -d '{"UserOne":3, "UserTwo":4}'
{}
> curl -X GET localhost:8000/api/match/1
{"ID":1,"UserOne":3,"UserTwo":0,"MatchedOn":<TIME>}
> curl -X DELETE localhost:8000/api/match/1
{}
> curl -X GET localhost:8000/api/match/1
{"error":"match not found with ID 1"}
```

Missing fields:
```
> curl -X POST localhost:8000/api/match -d '{"UserOne":5}'
{"error":"Missing request parameter(s)"}
> curl -X POST localhost:8000/api/match -d '{"UserOne":5, "UserTwo":6}'
{}
> curl -X PUT localhost:8000/api/match/1 -d '{"UserOne":7}'
{"error":"Missing request parameter(s)"}
```

### CODE REVIEW

File: match/match.go [hunk 1]
```diff
@@ -52,6 +52,36 @@ func jsonMiddleware(next http.Handler) http.Handler {
...
 
+func matchCreateHandler(w http.ResponseWriter, r *http.Request) {
+	var req matchDAO.MatchCreateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	if req.UserOne == nil || req.UserTwo == nil {
```
Comment (willjvsmith) [inline match/match.go:63]: Rather than doing this, I was thinking of writing a function that takes any struct and checks if any of the pointers are `nil`. This would make generation easier, but perhaps at the cost of readability.

I looked into this and it seems achievable [using reflection](https://stackoverflow.com/questions/49466522/how-to-check-if-all-fields-of-a-struct-are-nil). I thought I would get others' thoughts before going for it though. Lewis and I suspect Charlie will be all for, but Jay against.

File: match/dao/match-dao.go [hunk 2]
```diff
@@ -102,6 +98,30 @@ func (dao *DAO) GetMatch(id int64) (*MatchGetResponse, error) {
 	return &match, nil
 }
 
+// UpdateMatch updates an already existing match to two user IDs
+func (dao *DAO) UpdateMatch(matchID int64, request MatchUpdateRequest) error {
+	rowsAffected, err := executeQuery(dao.DB, "UPDATE Match SET userOne = $1, userTwo = $2 WHERE id = $3", request.UserOne, request.UserTwo, matchID)
```
Comment (jaylees14) [inline match/dao/match-dao.go]: Probably update the `matchedOn` time here too

File: match/match.go [hunk 3]
```diff
@@ -34,10 +34,10 @@ func main() {
 	}
 
 	r := mux.NewRouter()
-	r.HandleFunc("/match/{id}", matchGetHandler).Methods(http.MethodGet)
 	r.HandleFunc("/match", matchCreateHandler).Methods(http.MethodPost)
+	r.HandleFunc("/match/{id}", matchGetHandler).Methods(http.MethodGet)
 	r.HandleFunc("/match/{id}", matchDeleteHandler).Methods(http.MethodDelete)
-	r.HandleFunc("/match", matchUpdateHandler).Methods(http.MethodPatch)
+	r.HandleFunc("/match/{id}", matchUpdateHandler).Methods(http.MethodPut)
 	r.HandleFunc("/matches/{id}", matchListHandler).Methods(http.MethodGet)
```
Comment (jaylees14) [inline match/match.go:41]: Should also update this to `/match/all`

File: match/match.go (hunk 1 above)
Comment (jaylees14) [reply match/match.go:63]: I think this is fine

Comment (jaylees14) [reply match/match.go:63]: https://www.calhoun.io/how-to-determine-if-a-json-key-has-been-set-to-null-or-not-provided/ Could be a solution, although might again be less readable: although, would avoid using reflection ;)

**Review by jaylees14 — APPROVED** [review]
LGTM

File: match/match.go (hunk 3 above)
Comment (willjvsmith) [reply match/match.go:41]: Will do in new PR.

**Review by c-harding — APPROVED** [review]
looks good
